org.mwc.debrief.legacy/src/Debrief/Tools/Palette/CreateLabel.java

```diff
@@ -224,6 +230,9 @@ public final void undo()
      */
     public final void execute()
     {
+      if(addedLayer!=null) {

```

Comment:
@ayeshare - _I think_ we need to check if the layer is already present before adding it.  For `redo` I know it will have been deleted already, and we have to add it.  But, for the initial `execute` I'm pretty sure we've already added the new layer to the `Layers` object.

Reply:
Thanks @IanMayo